crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -122,6 +124,22 @@ impl SuiEvent {
     }
 }
 
+impl Display for SuiEvent {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let output = format!(
+            " ┌──\n │ EventID: {}\n │ PackageID: {}\n │ Transaction Module: {}\n │ Sender: {} \n │ EventType: {}:{}\n │ ParsedJSON: {}\n │ BCS: {:?}\n",
+            self.id.tx_digest, self.id.event_seq, self.package_id, self.transaction_module, self.sender, self.type_, self.parsed_json, self.bcs);
+        match self.timestamp_ms {
+            Some(ts) => {
+                writeln!(f, "{output} │ Timestamp: {}\n └──", ts)
+            }
+            None => {
+                writeln!(f, "{output} └──")
+            }
+        }
```

Comment:
I would probably structure this slightly differently, to have a pattern that works for multiple optional fields.

```suggestion
        if let Some(ts) = self.timestamp_ms {
            writeln!(f, "{output} │ Timestamp: {}\n └──", ts)?;
        }
        
        writeln!(f, "└──")
```